File: services/assesment-service/src/main/java/sgptt/assessvc/assesmentservice/entity/Activity.java
package sgptt.assessvc.assesmentservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import sgptt.assessvc.assesmentservice.model.ActivityName;

import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Activity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "activity_id")
    private Long activityId;

    @Temporal(TemporalType.DATE)
    private Date openDate;

    @Temporal(TemporalType.DATE)
    private Date closeDate;

    @Enumerated(EnumType.ORDINAL)
    private ActivityName activity;
}

File: services/auth-service/src/main/kotlin/com/sgpttt/authservice/repository/entity/Activity.kt
package com.sgpttt.authservice.repository.entity

import com.sgpttt.authservice.model.domain.ActivityName
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Temporal
import jakarta.persistence.TemporalType
import java.util.Date

@Entity
data class Activity(
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "activity_id")
	val activityId: Long,
	
	@Temporal(TemporalType.DATE)
	val openDate: Date,
	
	@Temporal(TemporalType.DATE)
	val closeDate: Date,
	
	@Enumerated(EnumType.ORDINAL)
	val activity: ActivityName,
)
